### ISSUE TITLE
Sequence-classification recipe over-scales gradients by dp_size

### ISSUE THREAD

**Issue by kabirvashisht4-glitch** [NVIDIA-NeMo/Automodel#3846]
**Describe the bug**

Identified by @yuhezhang-ai while reviewing #3830 — filing it so it is tracked
outside that PR thread.

`TrainFinetuneRecipeForSequenceClassification` multiplies the loss by `dp_size`
before `backward()`, but the loss is a **local** per-micro-batch mean. That
factor is only valid when the loss is already normalized by a **global**
denominator, so on `dp_size > 1` the gradient is over-scaled by `dp_size` — on
top of the separate `grad_accumulation_steps` factor in #3829.

`nemo_automodel/recipes/llm/train_seq_cls.py` L90 and L242:

```python
self.loss_fn = torch.nn.CrossEntropyLoss()     # local mean
...
(loss * self._get_dp_group_size(include_cp=True)).backward()
```

DDP/FSDP **average** gradients across DP ranks, so:

```
(1/dp) * sum_r [ dp * local_mean_r ] = sum_r local_mean_r = dp x global_mean    (wrong)
```

`recipes/llm/train_ft.py` uses the same `* dp_size` at L1193 and is correct,
because its loss is divided by `num_label_tokens`, which is DP-allreduced
(L1211-1213):

```
(1/dp) * sum_r [ dp * local_sum_r / N_global ] = global_sum / N_global          (right)
```

So `* dp_size` is not wrong in itself — it cancels the DP average. It is only
correct alongside a global denominator, and `train_seq_cls.py` does not have one.

**Steps/Code to reproduce bug**

CPU only. Simulates the DP gradient average against the true global-mean
gradient (4 DP ranks x 2 accumulation steps x 2 samples):

```python
import torch, torch.nn as nn, torch.nn.functional as F

torch.manual_seed(0)
V, C, DP, ACC, B = 8, 3, 4, 2, 2
lin = nn.Linear(V, C, bias=False)
W = lin.weight.detach().clone()
N = DP * ACC * B
X, Y = torch.randn(N, V), torch.randint(0, C, (N,))
shards = [[(X[(r*ACC+a)*B:(r*ACC+a+1)*B], Y[(r*ACC+a)*B:(r*ACC+a+1)*B]) for a in range(ACC)] for r in range(DP)]

def grad(fn):
    lin.weight.data.copy_(W); lin.weight.grad = None
    fn(); return lin.weight.grad.clone()

truth = grad(lambda: F.cross_entropy(lin(X), Y, reduction="mean").backward())

def ddp(per_rank):                       # DDP averages the per-rank grads
    gs = []
    for r in range(DP):
        lin.weight.data.copy_(W); lin.weight.grad = None
        per_rank(r); gs.append(lin.weight.grad.clone())
    return torch.stack(gs).mean(0)

def today(r):                            # what the recipe does
    for xb, yb in shards[r]:
        (F.cross_entropy(lin(xb), yb, reduction="mean") * DP).backward()

def train_ft_convention(r):              # local SUM over a GLOBAL denominator
    for xb, yb in shards[r]:
        (F.cross_entropy(lin(xb), yb, reduction="sum") / N * DP).backward()

print((ddp(today).norm() / truth.norm()).item())                 # 8.0
print((ddp(train_ft_convention).norm() / truth.norm()).item())   # 1.0
```

Driving the recipe's real `_run_train_optim_step` gives the same factors:

| `dp_size` | `grad_acc` | gradient vs. global mean |
|---:|---:|---:|
| 1 | 1 | 1.00x |
| 1 | 4 | 4.00x |
| 2 | 1 | **2.00x** |
| 2 | 2 | **4.00x** |
| 4 | 2 | **8.00x** |

The factor is exactly `dp_size * grad_accumulation_steps`. On 8 GPUs with
`grad_accumulation_steps=4` that is 32x — an effectively multiplied learning
rate, with no error and an unaffected logged loss, since only the gradient is
wrong.

Both factors are 1 at `dp_size=1, grad_acc=1`, so a single-GPU smoke test passes.

**Expected behavior**

The DP-averaged gradient equals the gradient of the mean loss over the global
batch, independent of `dp_size` and of how the batch is split into
micro-batches — matching `train_ft.py`.

**Environment overview**

- `main` at 3ddef9b1, CPU only; the DP average is simulated, no multi-GPU needed.

**Additional context**

Fixing only the accumulation half (#3829) leaves the `dp_size` half in place —
the middle row of the table above. Both come from the same root cause, so the
single fix is to adopt the `train_ft.py` convention: normalize by a DP-allreduced
global label count and keep `* dp_size`. I have pushed that to #3830, which now
supersedes its original narrower change and closes both this and #3829.

As @yuhezhang-ai noted on #3830, learning rates in existing sequence-classification
recipes may need restating afterwards. Worth flagging that the correction is not
uniform: each run absorbed a `dp_size * grad_acc` factor specific to its own
parallelism and batch config, so a learning rate tuned under one configuration
will not transfer to another. I do not have multi-GPU hardware to run those
experiments.
